polymesh-docs/polymesh-private/confidential-assets/diagrams.md

```diff
@@ -201,13 +196,19 @@ For an issuer to be able to be able to verify or decrypt confidential amounts th
 ```mermaid
 sequenceDiagram
 
+participant I as Issuer/Sender
+participant V as Venue Owner
 participant A as Auditor(s)
 participant P as Polymesh
-  A ->> P: Query previously emitted instruction affirmation events to retrieve<br>sender proof(s). A chain indexer (e.g. Polymesh SubQuery)<br> may be required for efficient queries of historical events
-  P -->> A: Sender proof
-  A ->> P: Query instruction leg details to determine the correct index of<br>the auditor confidential account in the array of auditors
-  P -->> A: Auditor index
-  A ->> A: Verify or decrypt amount from the sender proof using<br>their confidential account, the confidential asset ID<br>and the auditor index for their confidential account.<br>If known, the expected amount can be<br>provider to reduce the time to verify.
+alt Venue provides instruction ID
+  V ->> A: Instruction ID
+else Sender provides instruction ID
+  I ->> A: Instruction ID
+else Venue/Sender does not provide instruction ID
+  A ->>P: Requests pending instructions
```

Comment:
The chain does not have a method for auditors to get pending instructions. Subquery is required to get instructions an auditor is able to decrypt the confidential amount from. those instructions don't need to be pending they can also already be executed.